cli/src/git_util.rs

```diff
@@ -158,15 +158,16 @@ pub fn with_remote_git_callbacks<T>(
         .map(|x| x as &mut dyn FnMut(&git::Progress));
     // Based on Git's implementation: https://github.com/git/git/blob/43072b4ca132437f21975ac6acc6b72dc22fd398/sideband.c#L178
     let mut sideband_progress_callback = None;
-    let mut scratch = Vec::new();
     if print_sideband_messages {
+        let ui = &ui;
```

Comment:
nit: you can also remove `Mutex::new(ui)`. Perhaps, these commits can be reordered usch that `&mut Ui` stuff can be simplified first.

Reply:
Ah, I didn't realize I could remove the mutex. Will do that and reorder the commits.